api/v1alpha1/function_types.go

```diff
@@ -30,15 +30,14 @@ import (
 type FunctionSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
-	// +kubebuilder:validation:Required
```

Comment:
I think `spec.name` is not required

Reply:
for Function CRD it should be non-required since it is duplicated with `.metadata.name`, but for Mesh CRD, which reuses `FunctionSpec` and uses the `.spec.name` in Mesh controller as well.